### DISCUSSION TITLE
422 Unprocessable Entity when trying to upload files

### DISCUSSION THREAD

**Discussion by sheyzi** [fastapi/fastapi#6536] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

class ProductBase(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    image: str
    slug: str

class ProductIn(ProductBase):
    pass


def create_product(db: Session, request: schemas.ProductIn, category_id: int):
    new_product = models.Product(**request.dict(), category_id=category_id)
    db.add(new_product)
    db.commit()
    db.refresh(new_product)
    return new_product

def save_file(file: UploadFile = File(...)):
    result = cloudinary.uploader.upload(file.file)
    url = result.get("url")
    return url

@router.post('/product/create')
async def create_product(*, file: UploadFile = File(None), request: schemas.ProductIn, category_id: int,  db: Session = Depends(get_db), current_user: schemas.UserOut = Depends(get_current_active_user)):
    product = storeview.get_category_by_slug(db, request.slug)
    if product:
        raise HTTPException(status.HTTP_400_BAD_REQUEST,
                            detail=f"Product with slug {request.slug} already exists!")
    request.image = storeview.save_file(file)
    return storeview.create_product(db, request, category_id)
```

### Description

Whenever i open the swagger ui and try creating a new product i keep getting `422 Unprocessable Entity` instead of a successful response

Note: It works alright without the files upload, it still fails even after removing everything concerning the processing of the file but with the fileupload parameter still available there

I also tried doing the request with curl and not just swagger ui it fails anyways

### Operating System

Linux

### FastAPI Version

0.68.0

### Python Version

3.8.10

**Comment by Kludex** [comment]
What do you have in the response body?

**Comment by SamuelHeath** [comment, marked as answer]
@sheyziSilver not an expert but was able to replicate this locally as well.
I think the issue is because you're trying to use `multipart/form-data` and `application/json` at the same time to encode the request body.

The documentation around files even points this out as being an issue at the bottom of [this](https://fastapi.tiangolo.com/tutorial/request-forms-and-files/#define-file-and-form-parameters) page.

I suggest either:
- converting the parameters `category_id` and `request` to use `Form()` 
- Base64 encode the file before uploading it especially if its an image
- If the above are not possible maybe two separate POST requests, one to create the product and one to upload, and save the image/file.

Hope this helps :+1:

**Comment by sheyzi** [comment]
> @sheyziSilver not an expert but was able to replicate this locally as well.
> I think the issue is because you're trying to use `multipart/form-data` and `application/json` at the same time to encode the request body.
> 
> The documentation around files even points this out as being an issue at the bottom of [this](https://fastapi.tiangolo.com/tutorial/request-forms-and-files/#define-file-and-form-parameters) page.
> 
> I suggest either:
> 
>     * converting the parameters `category_id` and `request` to use `Form()`
> 
>     * Base64 encode the file before uploading it especially if its an image
> 
>     * If the above are not possible maybe two separate POST requests, one to create the product and one to upload, and save the image/file.
> 
> 
> Hope this helps +1

Thanks for your response. It works now and the problem was like you stated
All i had to do was to stop receiving request and instead receive them as form parameters

So 
```python
def create_user(first_name: str = Form(...), file: UploadFile = File(...)):
    ...
```
